### PR TITLE
[NEURON][BUILD][MX] - update to sdk1.17.0

### PR DESCRIPTION
**Pull request by aws-vrnatham**
*GitHub Issue #, if available:*

Note: If merging this PR should also close the associated Issue, please also add that Issue # to the Linked Issues section on the right. 

### Description
Move to the latest version of neuron sdk 1.17.0.

### Tests run
**NOTE: If you are creating a PR for a new framework version, please ensure success of the standard, rc, and efa sagemaker remote tests by updating the dlc_developer_config.toml file:**

- [ ] Revision A: `sagemaker_remote_tests = "standard"`
- [ ] Revision B: `sagemaker_remote_tests = "rc"`
- [ ] Revision C: `sagemaker_remote_tests = "efa"`

**Additionally, please run the sagemaker local tests in at least one revision:**
- [ ] `sagemaker_local_tests = true`

## Label Checklist
- [ ] I have added the project label for this PR (*<project_name>* or "Improvement")

## PR Checklist
- [x ] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [ei/neuron/graviton] | [build] | [test] | [benchmark] | [ec2, ecs, eks, sagemaker]
- [x ] If the PR changes affects SM test, I've modified dlc_developer_config.toml in my PR branch by setting sagemaker_tests = true and efa_tests = true
- [ ] If this PR changes existing code, the change fully backward compatible with pre-existing code. (Non backward-compatible changes need special approval.)
- [ ] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [ ] (If applicable) I've documented below the tests I've run on the DLC image
- [ ] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [ ] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

## Pytest Marker Checklist
- [ ] (If applicable) I have added the marker `@pytest.mark.model("<model-type>")` to the new tests which I have added, to specify the Deep Learning model that is used in the test (use `"N/A"` if the test doesn't use a model)
- [ ] (If applicable) I have added the marker `@pytest.mark.integration("<feature-being-tested>")` to the new tests which I have added, to specify the feature that will be tested
- [ ] (If applicable) I have added the marker `@pytest.mark.multinode(<integer-num-nodes>)` to the new tests which I have added, to specify the number of nodes used on a multi-node test
- [ ] (If applicable) I have added the marker `@pytest.mark.processor(<"cpu"/"gpu"/"eia"/"neuron">)` to the new tests which I have added, if a test is specifically applicable to only one processor type

#### EIA/NEURON/GRAVITON Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `ei_mode = true`, `neuron_mode = true` or `graviton_mode = true`

#### Benchmark Testing Checklist
* When creating a PR:
- [ ] I've modified `dlc_developer_config.toml` in my PR branch by setting `benchmark_mode = true`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: mxnet/inference/docker/1.8/py3/sdk1.17.0/Dockerfile.neuron [hunk 1]
```diff
@@ -10,8 +10,7 @@ LABEL com.amazonaws.sagemaker.capabilities.accept-bind-to-port=true
 # https://docs.aws.amazon.com/sagemaker/latest/dg/build-multi-model-build-container.html
 LABEL com.amazonaws.sagemaker.capabilities.multi-models=true
 
-ARG MMS_VERSION=1.1.4
-ARG MX_URL=https://aws-mx-pypi.s3-us-west-2.amazonaws.com/1.8.0/aws_mx-1.8.0-py2.py3-none-manylinux2014_x86_64.whl
+ARG MMS_VERSION=1.1.8
```
Comment (saimidu) [inline mxnet/inference/docker/1.8/py3/sdk1.17.0/Dockerfile.neuron:13]: nit: Why is line 106 on this dockerfile needed? I don't think it has any effect on the resultant docker image, because it doesn't upgrade the MMS version from the version defined in this ARG, but it is a redundant step.

File: test/sagemaker_tests/mxnet/inference/integration/sagemaker/test_neuron_hosting.py [hunk 2]
```diff
@@ -44,6 +44,7 @@ def test_neuron_hosting(sagemaker_session, ecr_image, instance_type, framework_v
         SCRIPT_PATH,
         framework_version=framework_version,
         image_uri=ecr_image,
+        model_server_workers=1,
```
Comment (saimidu) [inline test/sagemaker_tests/mxnet/inference/integration/sagemaker/test_neuron_hosting.py:47]: Why is this change necessary? Could you please add a comment here (if it is just a comment, CI tests aren't necessary) to explain why the number of workers needs to be hardcoded to 1?

If it is something that accompanies the upgrade to SDK 1.17.0, then please mention this in the PR Description for posterity.

**Review by saimidu — COMMENTED** [review]
Approved.

Will provide full approval once config toml file is reverted.

File: test/sagemaker_tests/mxnet/inference/integration/sagemaker/test_neuron_hosting.py [hunk 3]
```diff
@@ -38,12 +38,16 @@
 def test_neuron_hosting(sagemaker_session, ecr_image, instance_type, framework_version, skip_if_no_neuron):
     prefix = "mxnet-serving/neuron-handlers"
     model_data = sagemaker_session.upload_data(path=MODEL_PATH, key_prefix=prefix)
+    # Since fake inf is run in mnist-neuron.py and NEURONCORE_GROUP_SIZE env variable
+    # is not set, set the model server worker to 1 (or need to set env variable).
+    # Otherwise num workers defaults to 4 (4 cpu's) and run out of neuron cores
```
Comment (saimidu) [inline test/sagemaker_tests/mxnet/inference/integration/sagemaker/test_neuron_hosting.py:43]: Thank you for the explanation.

**Review by saimidu — APPROVED** [review]
